packages/std/src/init_handle.rs

```diff
@@ -169,6 +169,32 @@ where
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct MigrateResponse<T = Never>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    // let's make the positive case a struct, it contrains Msg: {...}, but also Data, Log, maybe later Events, etc.
+    pub messages: Vec<CosmosMsg<T>>,
```

Comment:
Do we really need to support message emission as part of the migration process? If we can remove the field here, we can get rid of the generic argument

Reply:
One clear case for governance migration is "burning" a contract. Basically, replacing it with a brick, but doing cleanup in the migrate step.

To properly clean up, you will often want to send all current bank balance to other users, so it is not lost forever. You could upload a contract that stores a set of claims, but maybe you just want to kill the contract and send all tokens to a given address. Without returning messages, you cannot do so.

This is my rationale at least, and my sample contract https://github.com/CosmWasm/cosmwasm/issues/387 would make use of this.

Reply:
Thanks for the explanation, makes sense